### ISSUE TITLE
[New build - DEFAULT] 22w06c: Player Progression and Reputation 

### ISSUE THREAD

**Issue by tsunamayo** [tsunamayo/Starship-EVO#4562]
This build introduces some player progression elements.
For now the work done was mainly on the UI side but more in-game features will be added.

**TAB-Codex Menu changes:**
- Tab menu has been re-organized: main tabs buttons move to the top, and sub-tab buttons move to the left side of the window.
This makes the navigation more hierarchical and logical while also allowing for further developments.
- A new Player Progression tab has been added, with three sub-tabs: Galactic Reputation, Consortium Reputation and Faction Reputation.

**Galactic Standing and Reputation:**
- the Galactic Standing is a xp score representing the player fame. It will increase as the player accomplish various actions in the game.
- For now only Landing at a landing pad (+10xp) and discovering a station (+50xp) and bounty mission gives some xp, but a lot more will be added.
- the Galactic Reputation is a score representing player honor or infamy. It can go negative if the player accomplishes some bad actions.
- The diplomatic alignment is unused at this stage, and will tie to faction play.

**Consortium and Faction Reputation:**
- Consortium reputation is a xp score increasing as the player accomplishes more action related to a particular consortium.
- Bounty mission increase the space police rank.
- Faction reputation is unused at this point.

**Still WIP:**
- Title and ranking system
- More player action yields xp
- Tie this system to Steam Achievements
- player bounty system

In a first time we will focus on making a compelling positive reputation play style, negative (aka pirate or outlaw) will come at a much later point.

**Hotfixes:**
#4548 Landing impossible in some cases.

**Comment by tsunamayo** [comment]
Build is up.
Another greedy Default push but I wanted the landing fix that could impact the tutorial. It was mainly UI work for now but I will add more way to gain xp and also some perk unlock. I will open a bunch of thread for player rank title and consortium, I am sorely in need of inspiration! Cheers

**Comment by Kaiser-Indrasil** [comment]
Nice to see some more gameplay elements!
We're drawing closer and closer to an actual survival mode.

Since it's a UI update, I would suggest changing the "Consortiums Reputation" to "Consortia Reputation" to better reflect the latin roots of this word.

![obraz](https://user-images.githubusercontent.com/63756057/153590209-a96439bd-02ad-4936-8aba-e94662fdf63a.png)

Also, "anarchy" would be a better fit here than "freedom".

Also, I think it's about time we picked an official currency name.
I vote Woolongs.

**Comment by Uncle-Ulty** [comment]
Imo it should be Order vs Upheaval, since you can have an order and freedom system, and an anarchist with order system. Upheaval seems to be a good antithesis for order.

Also, Woolong is an Intelectual Property of Cowboy Bebop franchise.

**Comment by Kaiser-Indrasil** [comment]
Upheaval, or chaos, either or.

@Uncle-Ulty Yeah, you're right.

**Comment by Uncle-Ulty** [comment]
Chaos is a good word for immediate understanding, but at the same time is a strong word. Upheaval is a softer concept compared to chaos, but it's a word that may seem hard to comprehend due to some language barriers... It depends on which effect Tsuna wants.

**Comment by cloudrambler** [comment]
What about "Turmoil", as it is similar to chaos in effect, but less severe.

**Comment by TwoNiner29** [comment]
> Nice to see some more gameplay elements! We're drawing closer and closer to an actual survival mode.
> 
> Since it's a UI update, I would suggest changing the "Consortiums Reputation" to "Consortia Reputation" to better reflect the latin roots of this word.
> 
> ![obraz](https://user-images.githubusercontent.com/63756057/153590209-a96439bd-02ad-4936-8aba-e94662fdf63a.png)
> 
> Also, "anarchy" would be a better fit here than "freedom".
> 
> Also, I think it's about time we picked an official currency name. I vote Woolongs.

Woolong is also the name of a chinese tea, although its spelt Oolong HAHA

**Comment by FlyingDebris** [comment]
I propose rolls of toilet paper and commemorative coffee mugs as the galactic unit of trade

Out of curiosity, are players going to be able to create their own factions with similar parameters later on down the line?

**Comment by tsunamayo** [comment]
On the Order vs Freedom I dont want something too negative instead of freedom, nor does it has to be the opposite of Order.
For Order think Empire and Freedom thing Rebellion. But I dont want something as simple as Evil vs Good.
It is like do you want more social liberty or a more cohesive and ordered society. It is kind of Right vs Left or more correctly conservatism vs progressivism, but I dont want to dig into politics as these days it is a minefield worse than Korea DMZ and it is likely very much culturally dependent.

Each faction will spawn somewhere on that spectrum which will give a starting reputation when a new Faction is encountered.

If you play Freedom you will still be able to be friend with an Order faction, up to a certain threshold point that is.

Once that threshold is met the name will change to reflect that being partisan change our vision of reality: ie it will be  Order vs Anarchy (for people playing Order), or Tyranny vs Freedom the other way around.

**Comment by tsunamayo** [comment]
> > Nice to see some more gameplay elements! We're drawing closer and closer to an actual survival mode.
> > Since it's a UI update, I would suggest changing the "Consortiums Reputation" to "Consortia Reputation" to better reflect the latin roots of this word.
> > ![obraz](https://user-images.githubusercontent.com/63756057/153590209-a96439bd-02ad-4936-8aba-e94662fdf63a.png)
> > Also, "anarchy" would be a better fit here than "freedom".
> > Also, I think it's about time we picked an official currency name. I vote Woolongs.
> 
> Woolong is also the name of a chinese tea, although its spelt Oolong HAHA

Yeah I always order Oolong tea that at the family restaurant. Or a beer.

**Comment by Kaiser-Indrasil** [comment]
> Once that threshold is met the name will change to reflect that being partisan change our vision of reality: ie it will be Order vs Anarchy (for people playing Order), or Tyranny vs Freedom the other way around.

Wow, that's honestly a very cool tidbit! I love when a game reflects the player's decisions in the smallest ways!
